core/src/main/scala/chisel3/Data.scala

```diff
@@ -436,55 +436,83 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
   }
 
   private[chisel3] def stringAccessor(chiselType: Option[String], vals: Option[String]): String = {
-    val (binding: String, enclosure: Option[BaseModule]) = bindingToString
-    s"${fmtParentStr}component${fmtSubcomponentStr(enclosure)}${binding}${fmtChiselTypeStr(chiselType)}${fmtValuesStr(vals)}"
+    if (this.isLit) { // TODO better polymorphic design?
+      (fmtChiselTypeStr(chiselType),
+        fmtValuesStr(vals)) match {
+        case (Some(a), Some(b)) => s"$a($b)"
+        case (None, None) => ""
+        case (Some(a), _) => s"$a"
+        case (_, Some(b)) => s"$b"
+      }
+    } else {
+      val (binding: Option[String], enclosure: Option[BaseModule]) = bindingToString
+        (fmtParentStr,
+          fmtSubcomponentStr(enclosure),
+          _computeName(None, None),
+          binding,
+          fmtChiselTypeStr(chiselType),
+          fmtValuesStr(vals)
+        ) match {
+        // TODO figure out which ones are redundant/impossible
+        //    parent   subcomp  binding  type     values
+        case (Some(a), Some(b), Some(name), Some(c), Some(d), Some(e)) => s"$a.$name.$b: $c[$d]($e)"
+        case (Some(a), Some(b), Some(name), Some(c), Some(d), None) => s"$a.$name.$b: $c[$d]"
+        case (Some(a), None, None, Some(c), Some(d), None) => s"$a: $c[$d]"
+        case (Some(a), Some(b), _, Some(c), Some(d), _) => s"$a.$b: $c[$d]"
+        case (Some(a), None, Some(name), Some(c), Some(d), None) => s"$a.$name: $c[$d]"
+        case (None, None, Some(name), Some(c), Some(d), Some(e)) => s"$c[$d]($e)"
+        case (Some(a), None, None, None, Some(d), None) => s"$d"
+        case (None, None, None, None, None, None) => s""
+        case (_, _, _, _, _, _) => s"" // TODO this should be deleted once all cases are accounted for
```

Comment:
can we make it an error for now to help find the cases that are not accounted for (or is this TODO going to be resolved in this PR)?

Reply:
Yeah I'm working on resolving this for this PR